File: API/Helpers/UserParams.cs

namespace API.Helpers
{
    public class UserParams:PaginationParams
    {
      
        public string CurrentUsername{set;get;}
        public string Gender{set;get;}
        
        public int MinAge{set;get;}=18;

        public int MaxAge{set;get;}=100;

        
        public string OrderBy{set;get;}="lastActive";

    }
}

File: Client/src/app/_models/userParams.ts
import { User } from "./User";

export class userParams{
    gender:string;
    minAge=18;
    maxAge=99;
     pageNumber=1;
    pageSize=12
    ;
    orderBy='lastActive';
    constructor(user:User)
    {
        this.gender=user.gender==='female'?'male':'female';
    }

}